model/models.go

```diff
@@ -41,3 +42,57 @@ type Request struct {
 	Log             *logrus.Entry
 	RequestTS       time.Time
 }
+
+// ResponseCache struct holding key and response for cache synchronization
+type ResponseCache struct {
```

Comment:
This structure is used as redis payload. I didn't use redis before, but I can imagine, that JSON has some overhead here: more network traffic and more memory consumption on redis side.
It the key attribute necessary here, it is kinda redundant to redis key? Would it make sense to use some binary format for encoding here to reduce the memory footprint?

Reply:
We could drastically reduce the payload if we only store Res in the prefixed key and clean it upon retrieval.
I think this would be sufficient information wise for an blocky instance to generate its cache entry.

My first idea was to also cache negative responses but later it occured to me that this would be a huge waste of resources.

I agree that currently that there is a huge overhead:
![image](https://user-images.githubusercontent.com/10324063/146335793-5940ec4f-7303-4c3d-92b1-fe72e88dcd16.png)

Reply:
Reason and RType are used only for logging purposes (also query log). Maybe we can remove them from caching and log in this case "reason: EXTERNAL_CACHE" or something similar